src/commonTest/kotlin/com/charleskorn/kaml/YamlReadingTest.kt

```diff
@@ -1305,6 +1305,99 @@ class YamlReadingTest : FlatFunSpec({
                 }
             }
 
+            context("restricting max alias count") {
+                val input = """
+                    .some-extension: &name Jamie
+
+                    members: [*name, *name]
+                """.trimIndent()
+
+                context("parsing anchors and aliases is disabled") {
+                    val configuration = YamlConfiguration(extensionDefinitionPrefix = ".", allowAnchorsAndAliases = false)
+                    val yaml = Yaml(configuration = configuration)
+
+                    test("throws an appropriate exception") {
+                        val exception = shouldThrow<ForbiddenAnchorOrAliasException> { yaml.decodeFromString(Team.serializer(), input) }
+
+                        exception.asClue {
+                            it.message shouldBe "Parsing anchors and aliases is disabled."
+                            it.line shouldBe 1
+                            it.column shouldBe 18
+                        }
+                    }
+                }
+
+                context("parsing anchors and aliases is enabled, max aliases count 0") {
+                    val configuration = YamlConfiguration(extensionDefinitionPrefix = ".", allowAnchorsAndAliases = true, maxAliasCount = 0)
+                    val yaml = Yaml(configuration = configuration)
+
+                    test("throws an appropriate exception") {
+                        val exception = shouldThrow<ForbiddenAnchorOrAliasException> { yaml.decodeFromString(Team.serializer(), input) }
+
+                        exception.asClue {
+                            it.message shouldBe "Parsing anchors and aliases is disabled."
+                            it.line shouldBe 1
+                            it.column shouldBe 18
+                        }
+                    }
+                }
+
+                context("parsing anchors and aliases is enabled, max aliases count 1") {
```

Comment:
Could you please also add a test case with max alias count set to 2, to verify that the content is allowed in this case?